File: typescript/src/n0232_implement_queue_using_stacks.test.ts
import { expect } from "jsr:@std/expect";
import { MyQueue } from "./n0232_implement_queue_using_stacks.ts";

Deno.test("MyQueue", () => {
  const queue = new MyQueue();
  queue.push(1);
  queue.push(2);
  expect(queue.peek()).toEqual(1);
  expect(queue.pop()).toEqual(1);
  expect(queue.empty()).toEqual(false);
});

File: rust/src/n0232_implement_queue_using_stacks.test.rs
use crate::n0232_implement_queue_using_stacks::MyQueue;

#[test]
fn test_queue() {
    let mut queue = MyQueue::new();
    queue.push(1);
    queue.push(2);
    assert_eq!(queue.peek(), 1);
    assert_eq!(queue.pop(), 1);
    assert!(!queue.empty())
}
